### PR TITLE
Add missing tests

### PR DESCRIPTION
**Pull request by svillegas-cdd**
* Added tests to `styles.ObjectCountHeaderPageNumberPagination`
* Added test to `styles.LinkHeaderPageNumberPagination`

Ref: https://cordada.aha.io/features/COMPCLDATA-253

### CODE REVIEW

**Review by fpinto-cdd — APPROVED** [review]

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 1]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        """
+        response = self.pagination_obj_1.get_paginated_response(self.data_list)
+        expected_headers = {
+            'Content-Type': 'text/html; charset=utf-8',
+            'Link': '<https://example.com/api/v1/endpoint/?page=2>; rel="next", '
+            '<https://example.com/api/v1/endpoint/?page=4>; rel="last"',
+        }
+        self.assertEqual(response.data, self.data_list)
+        self.assertEqual(response.headers, expected_headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:69]:
```suggestion
        self.assertEqual(response.status_code, http.HTTPStatus.OK)
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 2]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        self.assertEqual(response.data, self.data_list)
+        expected_headers = {
+            'Content-Type': 'text/html; charset=utf-8',
+            'Link': '<https://example.com/api/v1/endpoint/>; rel="first", '
+            '<https://example.com/api/v1/endpoint/>; rel="previous", '
+            '<https://example.com/api/v1/endpoint/?page=3>; rel="next", '
+            '<https://example.com/api/v1/endpoint/?page=4>; rel="last"',
+        }
+        self.assertEqual(response.headers, expected_headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:84]:
```suggestion
        self.assertEqual(response.status_code, http.HTTPStatus.OK)
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 3]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        """
+        response = self.pagination_obj_2.get_paginated_response([])
+
+        expected_headers = {
+            'Content-Type': 'text/html; charset=utf-8',
+            'Link': '',
+        }
+        self.assertEqual(response.data, [])
+        self.assertEqual(response.headers, expected_headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:98]:
```suggestion
        self.assertEqual(response.status_code, http.HTTPStatus.OK)
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 4]
```diff
@@ -1,8 +1,13 @@
 from __future__ import annotations
 
 import unittest
+from collections import OrderedDict
+from unittest import mock
 
 import rest_framework.pagination
+from django.core.paginator import Page, Paginator
+from rest_framework import status
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:9]:
```suggestion
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 5]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+    def test_get_paginated_response_with_items(self) -> None:
+        """
+        Test that the pagination response is correct.
+        """
+        response = self.pagination_obj_1.get_paginated_response(self.data_list)
+        expected_headers = {
+            'Content-Type': 'text/html; charset=utf-8',
+            'Link': '<https://example.com/api/v1/endpoint/?page=2>; rel="next", '
+            '<https://example.com/api/v1/endpoint/?page=4>; rel="last"',
+        }
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:66]: Reformat

```suggestion
        expected_headers = {
            'Content-Type': 'text/html; charset=utf-8',
            'Link': (
                '<https://example.com/api/v1/endpoint/?page=2>; rel="next", '
                '<https://example.com/api/v1/endpoint/?page=4>; rel="last"'
            ),
        }
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 6]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        self.pagination_obj_1.page = Page(self.paginator_1.get_page(2), 2, self.paginator_1)
+        response = self.pagination_obj_1.get_paginated_response(self.data_list)
+
+        self.assertEqual(response.data, self.data_list)
+        expected_headers = {
+            'Content-Type': 'text/html; charset=utf-8',
+            'Link': '<https://example.com/api/v1/endpoint/>; rel="first", '
+            '<https://example.com/api/v1/endpoint/>; rel="previous", '
+            '<https://example.com/api/v1/endpoint/?page=3>; rel="next", '
+            '<https://example.com/api/v1/endpoint/?page=4>; rel="last"',
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:81]: Reformat

```suggestion
            'Link': (
                '<https://example.com/api/v1/endpoint/>; rel="first", '
                '<https://example.com/api/v1/endpoint/>; rel="previous", '
                '<https://example.com/api/v1/endpoint/?page=3>; rel="next", '
                '<https://example.com/api/v1/endpoint/?page=4>; rel="last"'
            ),
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 7]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+                'collection_next_page': {
+                    'description': 'Next page of the collection of items.\n\n'
+                    'Use the URL returned in the relation type `next` of the response `Link` '
+                    'header.'
+                },
+                'collection_last_page': {
+                    'description': 'Last page of the collection of items.\n\n'
+                    'Use the URL returned in the relation type `last` of the response `Link` '
+                    'header.'
+                },
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:142]: Reformat

```suggestion
                'collection_first_page': {
                    'description': (
                    'First page of the collection of items.\n\n'
                    'Use the URL returned in the relation type `first` of the response `Link` '
                    'header.'
                    )
                },
                'collection_previous_page': {
                    'description': (
                    'Previous page of the collection of items.\n\n'
                    'Use the URL returned in the relation type `previous` of the response `Link` '
                    'header.'
                    )
                },
                'collection_next_page': {
                    'description': (
                    'Next page of the collection of items.\n\n'
                    'Use the URL returned in the relation type `next` of the response `Link` '
                    'header.'
                    )
                },
                'collection_last_page': {
                    'description': (
                    'Last page of the collection of items.\n\n'
                    'Use the URL returned in the relation type `last` of the response `Link` '
                    'header.'
                    )
                },
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 8]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+                    'examples': {
+                        'pagination': {
+                            'summary': 'Link Header Page Number Pagination.\n\n'
+                            'A simple page number–based style that supports page numbers as query '
+                            'parameters\nand includes pagination links in an RFC 8288–compliant '
+                            '`Link` header.',
+                            'value': '<http://www.example.com/example-items/>; rel="first", '
+                            '<http://www.example.com/example-items/?page=3>; rel="previous", '
+                            '<http://www.example.com/example-items/?page=5>; rel="next", '
+                            '<http://www.example.com/example-items/?page=42>; rel="last"',
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:157]: Reformat

```suggestion
                            'summary': (
                            'Link Header Page Number Pagination.\n\n'
                            'A simple page number–based style that supports page numbers as query '
                            'parameters\nand includes pagination links in an RFC 8288–compliant '
                            '`Link` header.',
                            'value': '<http://www.example.com/example-items/>; rel="first", '
                            '<http://www.example.com/example-items/?page=3>; rel="previous", '
                            '<http://www.example.com/example-items/?page=5>; rel="next", '
                            '<http://www.example.com/example-items/?page=42>; rel="last"'
                            ),
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 9]
```diff
@@ -12,6 +17,30 @@ class LinkHeaderPageNumberPaginationTest(unittest.TestCase):
...
+
+        # Pagination with 2 pages.
+        self.data_list = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+        self.paginator_1 = Paginator(self.data_list, 2)
+        self.pagination_obj_1 = LinkHeaderPageNumberPagination()
+        self.pagination_obj_1.request = mock.MagicMock(
+            build_absolute_uri=mock.Mock(
+                return_value='https://example.com/api/v1/endpoint/',
+            )
+        )
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:31]:
```suggestion
        self.pagination_obj_1.request = mock.create_autospec(
            rest_framework.request.Request,
            instance=True,
            build_absolute_uri=mock.Mock(
                return_value='https://example.com/api/v1/endpoint/',
            ),
        )
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 10]
```diff
@@ -12,6 +17,30 @@ class LinkHeaderPageNumberPaginationTest(unittest.TestCase):
...
+        self.pagination_obj_1.page = Page(self.paginator_1.get_page(1), 1, self.paginator_1)
+
+        # Empty Pagination with 1 page.
+        self.paginator_2 = Paginator([], 2)
+        self.pagination_obj_2 = LinkHeaderPageNumberPagination()
+        self.pagination_obj_2.request = mock.MagicMock(
+            build_absolute_uri=mock.Mock(
+                return_value='https://example.com/api/v1/endpoint/',
+            )
+        )
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:41]:
```suggestion
        self.pagination_obj_2.request = mock.create_autospec(
            rest_framework.request.Request,
            instance=True,
            build_absolute_uri=mock.Mock(
                return_value='https://example.com/api/v1/endpoint/',
            ),
        )
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 11]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        self.data_list = ['a', 'b', 'c', 'd']
+
+        # Pagination with 2 pages.
+        self.paginator_1 = Paginator(self.data_list, 2)
+        self.pagination_obj_1 = ObjectCountHeaderPageNumberPagination()
+        self.pagination_obj_1.request = mock.MagicMock(
+            build_absolute_uri=mock.Mock(
+                return_value='https://example.com/api/v1/endpoint/',
+            )
+        )
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:224]:
```suggestion
        self.pagination_obj_1.request = mock.create_autospec(
            rest_framework.request.Request,
            instance=True,
            build_absolute_uri=mock.Mock(
                return_value='https://example.com/api/v1/endpoint/',
            )
        )
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 12]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+        self.pagination_obj_1.page = Page(self.paginator_1.get_page(1), 1, self.paginator_1)
+
+        # Empty Pagination with 1 page.
+        self.paginator_2 = Paginator([], 1)
+        self.pagination_obj_2 = ObjectCountHeaderPageNumberPagination()
+        self.pagination_obj_2.request = mock.MagicMock(
+            build_absolute_uri=mock.Mock(
+                return_value='https://example.com/api/v1/endpoint/',
+            )
+        )
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:234]:
```suggestion
        self.pagination_obj_2.request = mock.create_autospec(
            rest_framework.request.Request,
            instance=True,
            build_absolute_uri=mock.Mock(
                return_value='https://example.com/api/v1/endpoint/',
            )
        )
```

File: fyntex/drf_pagination_utils/tests/test_styles.py [hunk 13]
```diff
@@ -25,16 +54,358 @@ def test_inherits_from_page_number_pagination_class(self) -> None:
...
+
+        # Pagination with 2 pages and object count header set.
+        self.paginator_3 = Paginator(self.data_list, 2)
+        self.pagination_obj_3 = ObjectCountHeaderPageNumberPagination()
+        self.pagination_obj_3.object_count_header = 'Count-Header'
+        self.pagination_obj_3.request = mock.MagicMock(
+            build_absolute_uri=mock.Mock(
+                return_value='https://example.com/api/v1/endpoint/',
+            )
+        )
```
Comment (jtrobles-cdd) [inline fyntex/drf_pagination_utils/tests/test_styles.py:245]:
```suggestion
        self.pagination_obj_3.request = mock.create_autospec(
            rest_framework.request.Request,
            instance=True,
            build_absolute_uri=mock.Mock(
                return_value='https://example.com/api/v1/endpoint/',
            )
        )
```

**Review by jtrobles-cdd — CHANGES_REQUESTED** [review]

**Review by jtrobles-cdd — APPROVED** [review]

**Pull request merged**